src/chains/tests/test_signals.py

```diff
@@ -0,0 +1,106 @@
+import os
+from unittest.mock import patch
+
+import responses
+from django.test import TestCase
+
+from chains.tests.factories import ChainFactory
+
+
+class ChainHookTestCase(TestCase):
+    @patch("requests.sessions.Session.post")
+    @patch.dict(os.environ, {})  # empty environment variables
+    def test_on_chain_update_with_no_cgw_set(self, mock_post):
+        ChainFactory.create()
+
+        assert not mock_post.called
+
+    @patch("requests.sessions.Session.post")
+    @patch.dict(os.environ, {"CGW_URL": "http://127.0.0.1"})
+    def test_on_chain_update_with_no_flush_token_set(self, mock_post):
+        ChainFactory.create()
+
+        assert not mock_post.called
+
+    @patch("requests.sessions.Session.post")
+    @patch.dict(
+        os.environ, {"CGW_URL": "http://127.0.0.1", "CGW_FLUSH_TOKEN": "example-token"}
+    )
+    def test_on_chain_create_hook_call(self, mock_post):
+        ChainFactory.create()
+
+        assert mock_post.called
+
+    @patch("requests.sessions.Session.post")
+    @patch.dict(
+        os.environ, {"CGW_URL": "http://127.0.0.1", "CGW_FLUSH_TOKEN": "example-token"}
+    )
+    def test_on_chain_delete_hook_call(self, mock_post):
+        chain = ChainFactory.create()
+
+        chain.delete()
+
+        # Call count should be 2: one for creation and one for deletion
+        self.assertEqual(mock_post.call_count, 2)
+
+    @patch("requests.sessions.Session.post")
```

Comment:
Why don't you use here `responses` library?

Reply:
I agree. One less mock.